src/createTokenBridge.ts

```diff
@@ -0,0 +1,213 @@
+  createTokenBridgeEnoughCustomFeeTokenAllowanceParamsOverride?: Partial<CreateTokenBridgeEnoughCustomFeeTokenAllowanceParams>;
+  createTokenBridgePrepareTransactionRequestParamsOverride?: Partial<CreateTokenBridgePrepareTransactionRequestParams>;
+  createTokenBridgePrepareRegisterWethGatewayTransactionRequestParamsOverride?: Partial<CreateTokenBridgePrepareRegisterWethGatewayTransactionRequestParams>;
+};
+
+/**
+ * Performs the transactions to deploy the token bridge core contracts
+ *
+ * For chain with custom gas token, it checks the custom gas token allowance
+ * token allowance and approve if necessary.
+ *
+ * Returns the token bridge core contracts.
+ *
+ * @param {CreateTokenBridgeParams} createTokenBridgeParams
+ * @param {Object} createRollupParams.rollupOwner - The rollup owner private key account
+ * @param {Object} createRollupParams.rollupAddress - The address of the Rollup contract
+ * @param {Object} createRollupParams.nativeTokenAddress - Optional
+ * If nativeTokenAddress is passed, deploy a token bridge with custom fee token.
+ * @param {Object} createRollupParams.parentChainPublicClient - The parent chain Viem Public Client
+ * @param {Object} createRollupParams.orbitChainPublicClient - The orbit chain Viem Public Client
+ * @param {Object} createRollupParams.createTokenBridgeEnoughCustomFeeTokenAllowanceParamsOverride - Optional {@link CreateTokenBridgeEnoughCustomFeeTokenAllowanceParams}
+ * @param {Object} createRollupParams.createTokenBridgePrepareTransactionRequestParamsOverride - Optional {@link CreateTokenBridgePrepareTransactionRequestParams}
+ * @param {Object} createRollupParams.createTokenBridgePrepareRegisterWethGatewayTransactionRequestParamsOverride - Optional {@link CreateTokenBridgePrepareSetWethGatewayTransactionRequestParams}
+ * @returns Promise<{@link TokenBridgeContracts}> - The token bridge core contracts.
+ *
+ * @example
+ * const tokenBridgeCreator = await deployTokenBridgeCreator({
+ *   publicClient: l2Client,
+ * });
+ *
+ * const tokenBridgeContracts = await createTokenBridge({
+ *   rollupOwner: rollupOwner,
+ *   rollupAddress: rollupAddress,
+ *   parentChainPublicClient: l1Client,
+ *   orbitChainPublicClient: l2Client,
+ *   createTokenBridgeEnoughCustomFeeTokenAllowanceParamsOverride: {
+ *     tokenBridgeCreatorAddressOverride: tokenBridgeCreator,
+ *   },
+ *   createTokenBridgePrepareTransactionRequestParamsOverride: {
+ *     tokenBridgeCreatorAddressOverride: tokenBridgeCreator,
+ *   },
+ *   createTokenBridgePrepareRegisterWethGatewayTransactionRequestParamsOverride: {
+ *     tokenBridgeCreatorAddressOverride: tokenBridgeCreator,
+ *     retryableGasOverrides: {
+ *       gasLimit: {
+ *         base: 100_000n,
+ *       }
+ *     }
+ *   }t
+ * });
+ */
+export async function createTokenBridge({
+  rollupOwner,
+  rollupAddress,
+  nativeTokenAddress,
+  parentChainPublicClient,
+  orbitChainPublicClient,
+  createTokenBridgeEnoughCustomFeeTokenAllowanceParamsOverride,
```

Comment:
I don't think this is necessary, you can just take the token bridge creator override from `createTokenBridgePrepareTransactionRequestParamsOverride`

Reply:
I would make the `tokenBridgeCreatorAddressOverride` on the top level of the object and just reuse it everywhere